cogs/Twitter.py

```diff
@@ -0,0 +1,693 @@
+from discord.ext import commands
+from cogs import CustomCog, AinitMixin
+from util import auto_help, ack
+
+logger = logging.getLogger(__name__)
+
+
+class TwitterAccountError(Exception):
+    """Not able to use twitter account exception"""
+
+    pass
+
+
+@events.priority(-15)
+def on_tweet_with_media(data):
+    """Event triggered when the data corresponds to a tweet with a media"""
+    return "media" in data.get("entities", {})
+
+
+def setup(bot):
+    bot.add_cog(Twitter(bot))
+
+
+def twitter_enabled():
+    async def predicate(ctx):
+        async with ctx.bot.Session() as session:
+            _server = await db.get_twitter_settings(session, ctx.guild.id)
+            return _server
+
+    return commands.check(predicate)
+
+
+class Twitter(CustomCog, AinitMixin):
+
+    FILESIZE_MAX = 8 * 10 ** 6  # 8 MB
+    URL_REGEX = r"(https?://)?(www.)?twitter.com/(\S+)/status/(\d+)(\?s=\d+)?"
+    EVENT_DATE_REGEX = r"(\s+|^)(\d{6}|\d{8})\s+"
+
+    def __init__(self, bot):
+        super().__init__(bot)
+
+        self.bot = bot
+        self.feed = None
+        self.stream_thing = None
+        self.client = None
+        self.stream_task = None
+        self.session = aiohttp.ClientSession()
+        self.keys = self.bot.config["cogs"]["twitter"]
+        TwtSorting.inject_bot(self.bot)
+        super(AinitMixin).__init__()
+
+    async def _ainit(self):
+        await self.bot.wait_until_ready()
+        self.client = PeonyClient(**self.keys, loop=self.bot.loop)
+        self.gen_accounts = await self.generate_accounts()
+        logger.info(f"Loaded {len(self.gen_accounts)} twitter accounts to follow")
+        if self.gen_accounts:
+            self.stream_task = asyncio.create_task(self.streaming())
+
+    async def generate_accounts(self):
+        async with self.bot.Session() as session:
+            accounts_list = await db.get_twitter_accounts(session)
+        # remove duplicate accounts
+        accounts_list = list(set([server.account_id for server in accounts_list]))
+        logger.info(
+            f"Generated new accounts list with {len(accounts_list)} twitter accounts to follow"
+        )
+        return accounts_list
+
+    async def streaming(self):
+        self.feed = self.client.stream.statuses.filter.post(follow=self.gen_accounts)
+
+        async with self.feed as stream:
+            self.stream_thing = stream
+            async for tweet in self.stream_thing:
+                if on_tweet_with_media(tweet):
+                    logger.info(
+                        f"stream processing @{tweet.user.screen_name} - {tweet.id_str}"
+                    )
+                    await self.manage_twt(tweet)
+
+    @auto_help
+    @commands.group(
+        name="twitter",
+        aliases=["twt"],
+        brief="Automatically post twitter pics to specific channels based on tags and accounts",
+        invoke_without_command=True,
+    )
+    async def twitter(self, ctx):
+        await ctx.send_help(self.twitter)
```

Comment:
```suggestion
    )
    async def twitter(self, ctx):
        if not ctx.invoked_subcommand:
            await ctx.send_help(self.twitter)
```

Change all the other groups too.